platforms/core-runtime/stdlib-java-extensions/src/main/java/org/gradle/api/JavaVersion.java

```diff
@@ -183,7 +184,9 @@ public enum JavaVersion {
      * @return The version, or null if the provided value is null.
      * @throws IllegalArgumentException when the provided value cannot be converted.
      */
+    @SuppressWarnings("NullAway") // We cannot annotate it as nullable as it would be a breaking change for Kotlin clients.
     public static JavaVersion toVersion(Object value) throws IllegalArgumentException {
+        //noinspection ConstantValue
         if (value == null) {
```

Comment:
💭 This is one of the places where we may want to deprecate the null argument to align Kotlin types with reality.